packages/fxa-shared/sentry/joi-message-overrides.ts

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { AnySchema } from 'joi';
+
+/**
+ * A set of default message overrides. These result in better error resolution in sentry.
+ */
+export const defaultMessageOverrides = {
+  // Override some of the message defaults. Here we remove the 'with value {:[.]}'
+  // portion of the message, because it causes too much fragmentation in our sentry
+  // errors. These should be applied to any .regex or .pattern joi validator
+  'string.pattern.base':
+    '{{#label}} fails to match the required pattern: {{#regex}}',
```

Comment:
@dschom I don't have a whole lot of context here besides checking out the filed issue but was taking a look to help move along some PRs - if this is exclusively for content-server, does it make more sense to live there instead of `fxa-shared` or is it something we should do everywhere? This looks like handlebars / mustache, but I know we're changing the joi validation messaging, so it looks a little odd but I'm sure it's g2g. (btw, this definitely seems more like a 2-pointer than 1-pointer)

Reply:
@LZoog Thanks for looking at this. I think this should be in fxa-shared as it is a cross cutting concern for all projects using joi. 

For a bit more context, see this file in the joi repo, https://github.com/hapijs/joi/blob/7aa36666863c1dde7e4eb02a8058e00555a99d54/lib/types/string.js#L718. Essentially they have their own error message templates. The issue with displaying the raw value is that it causes fragemenation in our sentry errors, i.e. each incorrect value will end up with a distinct error message which could be seen as a unique error to sentry, when really it's all the same validation issue. Values can also contain PII, so it's best not having them in error messages.

Reply:
Ah, thank you for the link, this makes a lot more sense to me now. 👍 It might be worth linking to their docs or something there after you mention "default joi message formats" for anyone squinting at it later since I didn't see any other similar references in the codebase, but I could have also realized I just needed to search for what that default format looks like, so 🤷‍♀️

Reply:
Definitely, I'll link to the their docs.